### PR TITLE
Updated to 6.4.4 -- added windows build

### PR DESCRIPTION
**Pull request by scharlton2**
@conda-forge-admin, please rerender

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
   version: {{ version }}
 
 source:
   url: https://github.com/MODFLOW-USGS/modflow6/releases/download/{{ version }}/mf{{ version }}_linux.zip
-  sha256: 8dcd832f33f992a9935c473fc3d695aac9250d4b34c944e54a7d44122df8e876
+  sha256: a737521126efc3295b818acc951a745d7b914446836c7931b0b52998f3934e2d
 
 build:
   number: 0
   skip: true  # [win]
```
Comment (mwtoews) [inline recipe/meta.yaml:14]: the only change needed to enable Windows is to remove this line

Comment (scharlton2) [reply recipe/meta.yaml:14]: I realized I left that in just after submitting.  I'll try again.

**Comment by jdhughes-dev** [comment]
@scharlton2 It does not look like the windows build is starting on Azure

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://github.com/MODFLOW-USGS/modflow6/releases/download/{{ version }}/mf{{ version }}_linux.zip
-  sha256: 8dcd832f33f992a9935c473fc3d695aac9250d4b34c944e54a7d44122df8e876
+  sha256: a737521126efc3295b818acc951a745d7b914446836c7931b0b52998f3934e2d
 
 build:
   number: 0
```
Comment (jdhughes-dev) [inline recipe/meta.yaml]: Update the build number from 0 -> 1.

Sorry for not merging the previous PR before now.

**Review by jdhughes-dev — COMMENTED** [review]
Work be good to add yourself (@scharlton2) as a maintainer.

File: recipe/meta.yaml (hunk 2 above)
Comment (scharlton2) [reply recipe/meta.yaml]: No problem

**Comment by scharlton2** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 3]
```diff
@@ -10,13 +10,12 @@ source:
...
-  number: 0
-  skip: true  # [win]
+  number: 1
 
 requirements:
   build:
-    - {{ compiler('fortran') }}
-    - {{ compiler('m2w64_fortran') }}  # [win]
+    - {{ compiler('fortran') }} # [not win]
+    - conda-forge/label/m2w64-experimental::mingw-w64-ucrt-x86_64-gcc-fortran # [win]
```
Comment (mwtoews) [inline recipe/meta.yaml]: Try restoring these to their original values, as it worked before (it was switched off with 316972ec23fa23ac826f9b2c33e3d37234a3e323). But I do see your attempt to use a more modern Fortran compiler for Windows (newer than [`m2w64-gcc-fortran` 5.3](https://anaconda.org/conda-forge/m2w64-gcc-fortran)).

**Comment by mwtoews** [comment]
I've just checked compiling the latest modflow6 code locally on a Windows computer with `m2w64-gcc-fortran`, and I see the same error on CI:
```
ninja: Entering directory `D:/src/modflow6/_build'
[13/448] Compiling Fortran object src/libmf6core.a.p/Utilities_CharString.f90.obj
FAILED: src/libmf6core.a.p/Utilities_CharString.f90.obj src/libmf6core.a.p/characterstringmodule.mod
"gfortran" "-Isrc\libmf6core.a.p" "-Isrc" "-I..\src" "-Isrc\libmf6_external.a.p" "-fdiagnostics-color=always" "-D_FILE_OFFSET_BITS=64" "-Wall" "-std=f2008" "-O0" "-g" "-fall-intrinsics" "-pedantic" "-cpp" "-Wcharacter-truncation" "-Wno-unused-dummy-argument" "-Wno-intrinsic-shadow" "-Wno-maybe-uninitialized" "-Wno-uninitialized" "-fcheck=all" "-ffpe-trap=overflow,zero,invalid" "-D_WIN32" "-Jsrc\libmf6core.a.p" -o src/libmf6core.a.p/Utilities_CharString.f90.obj "-c" ../src/Utilities/CharString.f90
f951.exe: internal compiler error: Segmentation fault
libbacktrace could not find executable to open
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://sourceforge.net/projects/msys2> for instructions.
[14/448] Compiling Fortran object src/libmf6core.a.p/Utilities_kind.f90.obj
ninja: build stopped: subcommand failed.
```
or simply with running a basic `gfortran` command:
```
D:\src\modflow6>gfortran -c src\Utilities\CharString.f90
f951.exe: internal compiler error: Segmentation fault
libbacktrace could not find executable to open
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://sourceforge.net/projects/msys2> for instructions.
```
This is a compiler bug. It might be possible to investigate and "fix" modflow6 to work with this old gfortran version for Windows.

Unfortunately, I'm not yet aware of a more modern Fortran version for Windows in conda-forge. The only other Windows compiler is flang, and I've not had much success using it.

File: recipe/meta.yaml (hunk 3 above)
Comment (scharlton2) [reply recipe/meta.yaml]: I'll give it a try

**Comment by mwtoews** [comment]
@conda-forge-admin, please rerender

**Comment by scharlton2** [comment]
Sorry, didn't see your rerender

**Comment by mwtoews** [comment]
Doesn't seem to matter, it's the same old gfortran compiler bug. However I can confirm that at local build can be made by using the newer Fortran compilers:

> mamba install conda-forge/label/m2w64-experimental::mingw-w64-ucrt-x86_64-gcc-fortran

this installs a modern `gfortran` (gcc 13.2.0) which works nicely with modflow6.

The challenge, which I see you've already made an attempt at, is to get this into the recipe...

**Comment by scharlton2** [comment]
I think the main problem is this:
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=881853&view=logs&j=a70f640f-cc53-5cd3-6cdc-236a1aa90802&t=6119ccbe-9301-594f-7c27-f792b80a7fcc&l=842

it's just too old (5.3.0 on windows vs 12.3.0 on linux/osx)

**Comment by scharlton2** [comment]
> Sorry again, just saw your latest comment

This worked on my local windows machine:

❯ micromamba env export --from-history
```
name: test2
channels:
- conda-forge
- conda-forge/label/m2w64-experimental
dependencies:
- conda-forge::meson
- conda-forge/label/m2w64-experimental::mingw-w64-ucrt-x86_64-gcc-fortran

```

But I couldn't figure out how to add the `conda-forge/label/m2w64-experimental` to the `meta.yaml` file

**Comment by mwtoews** [comment]
@conda-forge-admin, please rerender

**Comment by mwtoews** [comment]
Enabling the `conda-forge/label/m2w64-experimental` channel would be a good solution. I'm trying again, but I'm not sure if it's possible, see https://github.com/conda/conda-build/issues/532

**Comment by scharlton2** [comment]
We might be able to use `recipe/conda_build_config.yaml` to override channel resolution see [.ci_support/README](https://github.com/scharlton2/modflow6-feedstock/blob/main-with-windows-build/.ci_support/README) and [recipe/conda_build_config](https://github.com/conda-forge/scipy-feedstock/blob/9401aa4ee5d9a22c2c9d219e5fb6dbd580f30207/recipe/conda_build_config.yaml#L2)

**Comment by scharlton2** [comment]
@conda-forge-admin, please rerender

**Comment by scharlton2** [comment]
@jdhughes-usgs, @mwtoews, @scopatz, and @Hofer-Julian if there are no objections, I think this PR is ready to go.

**Comment by mwtoews** [comment]
I'm doing a few things locally. With this PR, I'm able to use a minimal environment with `conda-build` and `conda-verify`, then run:

    conda build path/to/recipe -c conda-forge/label/m2w64-experimental

(the `-c` channel arg is needed, otherwise an error is raised: "mingw-w64-ucrt-x86_64-gcc-fortran does not exist")

The local build and test appear to be successful. However,

    mamba install --dry-run modflow6

has error:
```
Could not solve for environment specs
The following packages are incompatible
├─ m2-conda-epoch is installable and it requires
│  └─ msys2-conda-epoch <0.0a0 , which can be installed;
└─ modflow6 is not installable because it requires
   └─ m2w64-gcc-libs, which requires
      └─ msys2-conda-epoch 20160418 , which conflicts with any installable versions previously reported.
```
It might not be possible to install this package without adding/modifying channels.

**Comment by scharlton2** [comment]
Thanks @mwtoews.  I'll try building locally tomorrow.

**Comment by scharlton2** [comment]
Sorry, just re-read your comment please ignore.
> 
> @mwtoews How did you build locally? Here's what I tried:
> 
> ```
> micromamba create -n modflow-local-build
> micromamba activate modflow-local-build
> micromamba install conda-build conda-verify
> micromamba activate modflow-local-build
> git clone git@github.com:scharlton2/modflow6-feedstock.git
> cd modflow6-feedstock
> git switch main-with-windows-build
> python .\build-locally.py
> ```
> 
> choosing `6. win_64_ config` yields:
> 
> ```
> selected win_64_
> Traceback (most recent call last):
>   File "C:\Users\charlton\source\repos\feedstocks\modflow6-feedstock\build-locally.py", line 106, in <module>
>     main()
>   File "C:\Users\charlton\source\repos\feedstocks\modflow6-feedstock\build-locally.py", line 87, in main
>     verify_config(ns)
>   File "C:\Users\charlton\source\repos\feedstocks\modflow6-feedstock\build-locally.py", line 61, in verify_config
>     raise ValueError(
> ValueError: only Linux/macOS configs currently supported, got win_64_
> ```

**Comment by scharlton2** [comment]
Replaced by #33

**Pull request closed without merging**